x-pack/plugins/enterprise_search/public/applications/app_search/components/search_ui/components/search_ui_form.tsx

```diff
@@ -0,0 +1,205 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useValues, useActions } from 'kea';
+
+import {
+  EuiForm,
+  EuiFormRow,
+  EuiSelect,
+  EuiComboBox,
+  EuiButton,
+  EuiComboBoxOptionOption,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { EngineLogic } from '../../engine';
+import { SearchUILogic } from '../search_ui_logic';
+import { ActiveField } from '../types';
+import { generatePreviwUrl } from '../utils';
+
+const TITLE_FIELD_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.titleFieldLabel',
+  { defaultMessage: 'Title field (Optional)' }
+);
+const TITLE_FIELD_HELP_TEXT = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.titleFieldHelpText',
+  { defaultMessage: 'Used as the top-level visual identifier for every rendered result' }
+);
+const TITLE_FIELD_PLACEHOLDER = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.titleFieldPlaceholder',
+  { defaultMessage: 'Select a title field' }
+);
+const FILTER_FIELD_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.filterFieldLabel',
+  { defaultMessage: 'Filter fields (Optional)' }
+);
+const FILTER_FIELD_HELP_TEXT = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.filterFieldHelpText',
+  { defaultMessage: 'Faceted values rendered as filters and available as query refinement' }
+);
+const FILTER_FIELD_PLACEHOLDER = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.filterFieldPlaceholder',
+  { defaultMessage: 'Click to select' }
+);
+const SORT_FIELD_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.sortFieldLabel',
+  { defaultMessage: 'Sort fields (Optional)' }
+);
+const SORT_FIELD_HELP_TEXT = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.sortHelpText',
+  { defaultMessage: 'Used to display result sorting options, ascending and descending' }
+);
+const SORT_FIELD_PLACEHOLDER = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.sortFieldPlaceholder',
+  { defaultMessage: 'Click to select' }
+);
+const URL_FIELD_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.urlFieldLabel',
+  { defaultMessage: 'URL field (Optional)' }
+);
+const URL_FIELD_HELP_TEXT = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.urlFieldHelpText',
+  { defaultMessage: "Used as a result's link target, if applicable" }
+);
+const URL_FIELD_PLACEHOLDER = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.urlFieldPlaceholder',
+  { defaultMessage: 'Select a URL field' }
+);
+const GENERATE_PREVIEW_BUTTON_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.generatePreviewButtonLabel',
+  { defaultMessage: 'Generate a Preview' }
```

Comment:
```suggestion
  { defaultMessage: 'Generate preview' }
```
Small copy suggestion - I was reading EUI's copy docs recently and apparently we should avoid "a" or "an" articles 🤔

<img width="1159" alt="" src="https://user-images.githubusercontent.com/549407/117191996-12f9ea00-ad96-11eb-826a-395a5c0a920a.png">

We have a lot of instances of those in our app though so I'm sure Chris will have a field day lol